# Results comparison for MIST with fine-tuned models of model A partial

In this notebook, I will compare the results of the different fine tuned models (only the MIST dataset).

In [1]:
from ML.utils.utils import *
from physics.Iso_data_handler import Iso_data_handler
from ML.utils.Data_preparator import Data_preparator

In [7]:
pre_path = "../../../../../../"
path_default = pre_path + "results/model_B/default_models"
path_fine_tuned = pre_path + "results/model_B/fine_tuned_models"

## Getting the ranges for the values

In [3]:
iso_handler = Iso_data_handler("C:/Users/antoi/Code/unif/MA2/thesis/data/MIST_v1.2_vvcrit0.0_basic_isos/", 
                                    ['log10_isochrone_age_yr', 'log_Teff', 'log_g', 'phase', 'metallicity', 'star_mass', 'log_R', 'log_L'], 
                                    "MIST", reclassify=True)

iso_df = iso_handler.get_isochrone_dataframe()
display(iso_df)

Reading MIST dataframe from MIST_reclassified_iso_full_data.csv file...


,log10_isochrone_age_yr,log_Teff,log_g,phase,metallicity,star_mass,log_R,log_L
0,5.0,3.486221,3.131342,-1.0,-0.25,0.100000,0.153402,-0.793667
1,5.0,3.487362,3.126808,-1.0,-0.25,0.102645,0.160326,-0.775254
2,5.0,3.489243,3.119367,-1.0,-0.25,0.107039,0.171785,-0.744810
3,5.0,3.491102,3.112165,-1.0,-0.25,0.111419,0.183099,-0.714749
4,5.0,3.492937,3.105143,-1.0,-0.25,0.115789,0.194305,-0.684996
...,...,...,...,...,...,...,...,...
1467117,10.3,4.402490,7.777159,6.0,0.50,0.532726,-1.806255,-1.047903
1467118,10.3,4.387132,7.783242,6.0,0.50,0.532730,-1.809295,-1.115415
1467119,10.3,4.371789,7.789130,6.0,0.50,0.532735,-1.812237,-1.182672
1467120,10.3,4.356480,7.794844,6.0,0.50,0.532741,-1.815091,-1.249617


In [4]:
phase_filtered_iso_df = Data_preparator.filter_data(iso_df, {'phase':[0, 2, 3], 'star_mass':("<", 30)})
print_all_uniques(phase_filtered_iso_df)

log10_isochrone_age_yr  : Range : 5.0 - 10.3, Mean : 8.2913, Median : 8.3

log_Teff  : Range : 3.34 - 5.3542, Mean : 3.9814, Median : 3.9163

log_g  : Range : -1.08 - 6.2434, Mean : 3.5356, Median : 3.9078

Values in phase column : 0.0, 2.0, 3.0 

Values in metallicity column : -0.25, -0.5, -0.75, -1.0, -1.25, -1.5, -1.75, -2.0, -2.5, -3.0, -3.5, -4.0, 0.0, 0.25, 0.5 

star_mass  : Range : 0.1 - 29.9991, Mean : 5.0421, Median : 2.422

log_R  : Range : -1.0 - 3.1298, Mean : 0.6593, Median : 0.6146

log_L  : Range : -3.33 - 5.9763, Mean : 2.1988, Median : 2.3277



## Default vs fine-tuned

### KNN

In [5]:
compare_metrics(path_default, value_rounding=3, relative_values=False,
                output_parameters=["T_eff", "log_g", "radius"],
                physical_models=["MIST"],
                model_names=["KNN"],
                data_filters=["scale_quantile_14"],
                categories=["Global"])

Results for the Global category with a value of 1.0


value range  \
model physical_model filter            output_parameter                   
KNN   MIST           scale_quantile_14 T_eff              3.345 - 4.634   
                                       log_g             -1.083 - 5.433   
                                       radius            -0.997 - 3.082   

                                                           RVE   RMSE    MAE  \
model physical_model filter            output_parameter                        
KNN   MIST           scale_quantile_14 T_eff             0.977  0.043  0.012   
                                       log_g             0.871  0.445  0.133   
                                       radius            0.900  0.222  0.067   

                                                         MedAE   CORR  MAX_ER  \
model physical_model filter            output_parameter                         
KNN   MIST           scale_quantile_14 T_eff             0.002  0.988   0.916   
                                       log_g             0.005  0.934   5.649   
                                       radius            0.004  0.949   2.823   

                                                                                                         Percentiles  \
model physical_model filter            output_parameter                                                                
KNN   MIST           scale_quantile_14 T_eff              75 : 0.00429 / 90 : 0.02469 / 95 : 0.06151 / 99 : 0.16181    
                                       log_g              75 : 0.01943 / 90 : 0.34454 / 95 : 0.97233 / 99 : 2.14109    
                                       radius             75 : 0.01134 / 90 : 0.17227 / 95 : 0.48608 / 99 : 1.07047    

                                                         time  
model physical_model filter            output_parameter        
KNN   MIST           scale_quantile_14 T_eff             5.66  
                                       log_g             5.66  
                                       radius            5.66

In [8]:
compare_metrics(path_fine_tuned, value_rounding=3, relative_values=False,
                output_parameters=["T_eff", "log_g", "radius"],
                physical_models=["MIST"],
                model_names=["KNN"],
                data_filters=["scale_quantile_14"],
                categories=["Global"])

Results for the Global category with a value of 1.0


value range  \
model physical_model filter            output_parameter                   
KNN   MIST           scale_quantile_14 T_eff              3.345 - 4.634   
                                       log_g             -1.083 - 5.433   
                                       radius            -0.997 - 3.082   

                                                           RVE   RMSE    MAE  \
model physical_model filter            output_parameter                        
KNN   MIST           scale_quantile_14 T_eff             0.978  0.042  0.013   
                                       log_g             0.881  0.426  0.138   
                                       radius            0.908  0.213  0.070   

                                                         MedAE   CORR  MAX_ER  \
model physical_model filter            output_parameter                         
KNN   MIST           scale_quantile_14 T_eff             0.002  0.989   0.875   
                                       log_g             0.008  0.939   5.652   
                                       radius            0.005  0.953   2.825   

                                                                                                         Percentiles  \
model physical_model filter            output_parameter                                                                
KNN   MIST           scale_quantile_14 T_eff               75 : 0.00718 / 90 : 0.03236 / 95 : 0.06481 / 99 : 0.1582    
                                       log_g              75 : 0.03594 / 90 : 0.38488 / 95 : 0.96954 / 99 : 1.99555    
                                       radius             75 : 0.02028 / 90 : 0.19245 / 95 : 0.48476 / 99 : 0.99773    

                                                          time  
model physical_model filter            output_parameter         
KNN   MIST           scale_quantile_14 T_eff             14.01  
                                       log_g             14.01  
                                       radius            14.01

the fine-tuning did not improve the results

### Random forest

In [9]:
compare_metrics(path_default, value_rounding=3, relative_values=False,
                output_parameters=["T_eff", "log_g", "radius"],
                physical_models=["MIST"],
                model_names=["random_forest"],
                data_filters=["scale_quantile_14"],
                categories=["Global"])

Results for the Global category with a value of 1.0


value range  \
model         physical_model filter            output_parameter                   
random_forest MIST           scale_quantile_14 T_eff              3.345 - 4.634   
                                               log_g             -1.083 - 5.433   
                                               radius            -0.997 - 3.082   

                                                                   RVE   RMSE  \
model         physical_model filter            output_parameter                 
random_forest MIST           scale_quantile_14 T_eff             0.967  0.052   
                                               log_g             0.814  0.533   
                                               radius            0.857  0.267   

                                                                   MAE  MedAE  \
model         physical_model filter            output_parameter                 
random_forest MIST           scale_quantile_14 T_eff             0.013  0.001   
                                               log_g             0.158  0.004   
                                               radius            0.079  0.002   

                                                                  CORR  \
model         physical_model filter            output_parameter          
random_forest MIST           scale_quantile_14 T_eff             0.984   
                                               log_g             0.905   
                                               radius            0.927   

                                                                 MAX_ER  \
model         physical_model filter            output_parameter           
random_forest MIST           scale_quantile_14 T_eff              0.887   
                                               log_g              5.653   
                                               radius             2.827   

                                                                                                                 Percentiles  \
model         physical_model filter            output_parameter                                                                
random_forest MIST           scale_quantile_14 T_eff              75 : 0.00385 / 90 : 0.02734 / 95 : 0.07453 / 99 : 0.19742    
                                               log_g              75 : 0.01869 / 90 : 0.36128 / 95 : 1.20554 / 99 : 2.67219    
                                               radius             75 : 0.00949 / 90 : 0.18062 / 95 : 0.60279 / 99 : 1.33623    

                                                                    time  
model         physical_model filter            output_parameter           
random_forest MIST           scale_quantile_14 T_eff             206.138  
                                               log_g             206.138  
                                               radius            206.138

In [10]:
compare_metrics(path_fine_tuned, value_rounding=3, relative_values=False,
                output_parameters=["T_eff", "log_g", "radius"],
                physical_models=["MIST"],
                model_names=["random_forest"],
                data_filters=["scale_quantile_14"],
                categories=["Global"])

Results for the Global category with a value of 1.0


value range  \
model         physical_model filter            output_parameter                   
random_forest MIST           scale_quantile_14 T_eff              3.345 - 4.634   
                                               log_g             -1.083 - 5.433   
                                               radius            -0.997 - 3.082   

                                                                   RVE   RMSE  \
model         physical_model filter            output_parameter                 
random_forest MIST           scale_quantile_14 T_eff             0.978  0.042   
                                               log_g             0.877  0.435   
                                               radius            0.905  0.217   

                                                                   MAE  MedAE  \
model         physical_model filter            output_parameter                 
random_forest MIST           scale_quantile_14 T_eff             0.014  0.004   
                                               log_g             0.146  0.015   
                                               radius            0.073  0.008   

                                                                  CORR  \
model         physical_model filter            output_parameter          
random_forest MIST           scale_quantile_14 T_eff             0.989   
                                               log_g             0.936   
                                               radius            0.951   

                                                                 MAX_ER  \
model         physical_model filter            output_parameter           
random_forest MIST           scale_quantile_14 T_eff              0.825   
                                               log_g              5.639   
                                               radius             2.821   

                                                                                                                 Percentiles  \
model         physical_model filter            output_parameter                                                                
random_forest MIST           scale_quantile_14 T_eff               75 : 0.01005 / 90 : 0.03141 / 95 : 0.06372 / 99 : 0.1596    
                                               log_g               75 : 0.05496 / 90 : 0.36198 / 95 : 0.9747 / 99 : 2.09301    
                                               radius             75 : 0.02792 / 90 : 0.18097 / 95 : 0.48731 / 99 : 1.04639    

                                                                    time  
model         physical_model filter            output_parameter           
random_forest MIST           scale_quantile_14 T_eff             266.507  
                                               log_g             266.507  
                                               radius            266.507

Improved the results of the extremes but not the middle values

## Comparison between fine tuned models

### T_eff parameter

In [14]:
compare_metrics(path_fine_tuned, value_rounding=3, relative_values=True,
                output_parameters=["T_eff"],
                physical_models=["MIST"],
                categories=["Global"],
                model_names=["KNN", "random_forest"])

Results for the Global category with a value of 1.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
KNN,MIST,scale_quantile_14,T_eff,3.345 - 4.634,0.978,3.253%,0.999%,0.167%,0.989,67.842%,75 : 0.557% / 90 : 2.509% / 95 : 5.025% / 99 : 12.265%,14.010
random_forest,MIST,scale_quantile_14,T_eff,3.345 - 4.634,0.978,3.266%,1.084%,0.272%,0.989,63.964%,75 : 0.779% / 90 : 2.435% / 95 : 4.94% / 99 : 12.374%,266.507


### log_g parameter

In [15]:
compare_metrics(path_fine_tuned, value_rounding=3, relative_values=True,
                output_parameters=["log_g"],
                physical_models=["MIST"],
                categories=["Global"],
                model_names=["KNN", "random_forest"])

Results for the Global category with a value of 1.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
KNN,MIST,scale_quantile_14,log_g,-1.083 - 5.433,0.881,6.545%,2.125%,0.13%,0.939,86.74%,75 : 0.552% / 90 : 5.907% / 95 : 14.879% / 99 : 30.625%,14.010
random_forest,MIST,scale_quantile_14,log_g,-1.083 - 5.433,0.877,6.669%,2.237%,0.233%,0.936,86.543%,75 : 0.843% / 90 : 5.555% / 95 : 14.959% / 99 : 32.121%,266.507


### Radius parameter

In [16]:
compare_metrics(path_fine_tuned, value_rounding=3, relative_values=True,
                output_parameters=["radius"],
                physical_models=["MIST"],
                categories=["Global"],
                model_names=["KNN", "random_forest"])

Results for the Global category with a value of 1.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
KNN,MIST,scale_quantile_14,radius,-0.997 - 3.082,0.908,5.229%,1.722%,0.131%,0.953,69.256%,75 : 0.497% / 90 : 4.717% / 95 : 11.883% / 99 : 24.457%,14.010
random_forest,MIST,scale_quantile_14,radius,-0.997 - 3.082,0.905,5.326%,1.797%,0.205%,0.951,69.146%,75 : 0.684% / 90 : 4.436% / 95 : 11.945% / 99 : 25.65%,266.507


The KNN model gives the best results.

## Comparison

| Parameter      | mass_filtered_smaller_14                                           |
| -------------  | ------------------------------------------------------------------ |
| age            | Range : 5.0 - 10.3, Mean : 8.4693, Median : 8.5                    |
| mass           | star_mass  : Range : 0.1 - 13.9995, Mean : 3.3831, Median : 2.0289 |
| log_teff       | log_Teff  : Range : 3.34 - 5.2138, Mean : 3.9383, Median : 3.867   |
| log_g          | log_g  : Range : -1.08 - 7.9405, Mean : 3.5574, Median : 3.9537    |
| log_R          | log_R  : Range : -1.86 - 3.0828, Mean : 0.6007, Median : 0.5357    |

### Mass parameter

In [3]:
compare_metrics(path, value_rounding=3, relative_values=False,
                output_parameters=["mass"],
                physical_models=["MIST"],)

Results for the Global category with a value of 1.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,mass,0.1 - 13.999,1.0,0.051,0.016,0.003,1.0,5.054,75 : 0.01204 / 90 : 0.03835 / 95 : 0.07323 / 99 : 0.22204,17.916
KNN,MIST,mass_smaller_14,mass,0.1 - 13.999,1.0,0.058,0.020,0.005,1.0,2.562,75 : 0.01728 / 90 : 0.0509 / 95 : 0.08791 / 99 : 0.2393,11.112
random_forest,MIST,mass_smaller_14,mass,0.1 - 13.999,1.0,0.031,0.011,0.003,1.0,1.852,75 : 0.00931 / 90 : 0.02399 / 95 : 0.04372 / 99 : 0.13158,1986.565


Results for the phase category with a value of 0.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,mass,0.1 - 13.999,1.0,0.051,0.017,0.004,1.0,5.054,75 : 0.01362 / 90 : 0.04204 / 95 : 0.07397 / 99 : 0.19481,17.916
KNN,MIST,mass_smaller_14,mass,0.1 - 13.999,1.0,0.063,0.021,0.006,1.0,2.562,75 : 0.0181 / 90 : 0.04796 / 95 : 0.0786 / 99 : 0.2504,11.112
random_forest,MIST,mass_smaller_14,mass,0.1 - 13.999,1.0,0.025,0.007,0.002,1.0,1.852,75 : 0.00611 / 90 : 0.01564 / 95 : 0.02668 / 99 : 0.07487,1986.565


Results for the phase category with a value of 2.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,mass,0.654 - 13.928,1.0,0.025,0.008,0.000,1.0,1.411,75 : 0.00578 / 90 : 0.02328 / 95 : 0.03465 / 99 : 0.08609,17.916
KNN,MIST,mass_smaller_14,mass,0.654 - 13.928,1.0,0.031,0.012,0.003,1.0,0.643,75 : 0.01036 / 90 : 0.03343 / 95 : 0.06162 / 99 : 0.13788,11.112
random_forest,MIST,mass_smaller_14,mass,0.654 - 13.928,1.0,0.019,0.009,0.005,1.0,0.463,75 : 0.01036 / 90 : 0.02108 / 95 : 0.03297 / 99 : 0.08395,1986.565


Results for the phase category with a value of 3.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,mass,0.545 - 13.994,0.999,0.074,0.027,0.003,1.0,2.379,75 : 0.01668 / 90 : 0.07299 / 95 : 0.14939 / 99 : 0.34725,17.916
KNN,MIST,mass_smaller_14,mass,0.545 - 13.994,1.000,0.072,0.031,0.006,1.0,1.351,75 : 0.02859 / 90 : 0.09174 / 95 : 0.15361 / 99 : 0.3117,11.112
random_forest,MIST,mass_smaller_14,mass,0.545 - 13.994,1.000,0.053,0.024,0.007,1.0,0.713,75 : 0.02273 / 90 : 0.0662 / 95 : 0.10988 / 99 : 0.23824,1986.565


### Radius parameter

In [11]:
compare_metrics(path, value_rounding=3,
                output_parameters=["radius"],
                physical_models=["MIST"],)

Results for the Global category with a value of 1.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,radius,-0.997 - 3.082,1.0,0.009,0.005,0.003,1.0,0.446,75 : 0.00675 / 90 : 0.01225 / 95 : 0.01613 / 99 : 0.03482,17.916
KNN,MIST,mass_smaller_14,radius,-0.997 - 3.082,1.0,0.005,0.003,0.002,1.0,0.179,75 : 0.00346 / 90 : 0.00661 / 95 : 0.00922 / 99 : 0.01696,11.112
random_forest,MIST,mass_smaller_14,radius,-0.997 - 3.082,1.0,0.004,0.002,0.001,1.0,0.117,75 : 0.00227 / 90 : 0.00472 / 95 : 0.00715 / 99 : 0.01498,1986.565


Results for the phase category with a value of 0.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,radius,-0.997 - 3.082,1.0,0.006,0.004,0.002,1.0,0.311,75 : 0.00456 / 90 : 0.00807 / 95 : 0.01132 / 99 : 0.02404,17.916
KNN,MIST,mass_smaller_14,radius,-0.997 - 3.082,1.0,0.004,0.003,0.002,1.0,0.057,75 : 0.00279 / 90 : 0.00559 / 95 : 0.00795 / 99 : 0.01523,11.112
random_forest,MIST,mass_smaller_14,radius,-0.997 - 3.082,1.0,0.002,0.001,0.001,1.0,0.045,75 : 0.00115 / 90 : 0.00231 / 95 : 0.00342 / 99 : 0.00636,1986.565


Results for the phase category with a value of 2.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,radius,0.091 - 2.871,0.999,0.011,0.007,0.006,1.0,0.446,75 : 0.00984 / 90 : 0.01384 / 95 : 0.01724 / 99 : 0.03213,17.916
KNN,MIST,mass_smaller_14,radius,0.091 - 2.871,1.000,0.004,0.003,0.002,1.0,0.064,75 : 0.00419 / 90 : 0.00709 / 95 : 0.00906 / 99 : 0.01321,11.112
random_forest,MIST,mass_smaller_14,radius,0.091 - 2.871,1.000,0.003,0.002,0.002,1.0,0.046,75 : 0.00318 / 90 : 0.00548 / 95 : 0.00736 / 99 : 0.01169,1986.565


Results for the phase category with a value of 3.0


,,,,value range,RVE,RMSE,MAE,MedAE,CORR,MAX_ER,Percentiles,time
model,physical_model,filter,output_parameter,,,,,,,,,
decision_tree,MIST,mass_smaller_14,radius,0.076 - 3.064,0.999,0.014,0.007,0.004,1.0,0.212,75 : 0.00865 / 90 : 0.01704 / 95 : 0.02612 / 99 : 0.05948,17.916
KNN,MIST,mass_smaller_14,radius,0.076 - 3.064,1.000,0.008,0.004,0.002,1.0,0.179,75 : 0.00438 / 90 : 0.00871 / 95 : 0.01276 / 99 : 0.03767,11.112
random_forest,MIST,mass_smaller_14,radius,0.076 - 3.064,1.000,0.008,0.004,0.002,1.0,0.117,75 : 0.00507 / 90 : 0.01006 / 95 : 0.01438 / 99 : 0.0308,1986.565


Random forest is the best model from all the fine tuned ones